terragrunt/aws/cloudfront/waf.tf

```diff
@@ -0,0 +1,334 @@
+resource "aws_wafv2_web_acl" "api_waf" {
+  provider = aws.ca-central-1
+
+  name        = "api_waf"
+  description = "WAF for API protection"
+  scope       = "CLOUDFRONT"
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+
+  default_action {
+    allow {}
+  }
+
+  rule {
+    name     = "IpAllowList"
+    priority = 1
+
+    action {
+      allow {}
+    }
+
+    statement {
+      ip_set_reference_statement {
+        arn = aws_wafv2_ip_set.ip_allowlist.arn
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "IpAllowList"
+      sampled_requests_enabled   = true
+    }
+  }
```

Comment:
This rule can also go since it was added to another project that was geo limiting to North America.